Mathlib/Order/Interval/Set/Icc.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2025 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll, Kevin Buzzard, Bhavik Mehta, Oliver Nash
+-/
+import Mathlib.Data.Fintype.Order
+import Mathlib.Data.Nat.Lattice
+import Mathlib.Data.Int.ConditionallyCompleteOrder
+import Mathlib.Data.Int.Interval
+import Mathlib.Data.Int.SuccPred
+
+/-!
+# Characterisation of the closed interval
+
+In this file we provide a characterisation of the closed interval for locally-finite conditionally
+complete linear orders, together with convenience specialisations for use with types such as
+`ℕ`, `ℤ`, `Fin n`.
+
+## Main results:
+ * `Set.Nonempty.eq_Icc_of_Ioo_subseteq_compl_empty`: a characterisation of closed intervals for
+   locally-finite conditionally complete linear orders.
+ * `Set.Nonempty.eq_Icc_of_Ioo_subseteq_compl_empty'`: characterisation of closed intervals for
+   locally-finite conditionally complete linear orders (convenient for `Fin n`).
+ * `Set.Finite.eq_Icc_of_Ioo_subseteq_compl_empty_nat`: characterisation of closed intervals
+   for `ℕ`.
+ * `Set.Finite.eq_Icc_of_Ioo_subseteq_compl_empty_int`: characterisation of closed intervals
+   for `ℤ`.
+-/
+
+variable {α : Type*} {I : Set α}
+
+lemma Set.Finite.ordConnected_of_Ioo_subseteq_compl_empty [LinearOrder α]
+    (h₁ : I.Finite)
+    (h₂ : ∀ᵉ (x ∈ I) (y ∈ I), Ioo x y ⊆ Iᶜ → Ioo x y = ∅) :
```

Comment:
Is `Ioo x y = ∅` better written as `y ≤ x` (along with a simp lemma to make the transformation?)

Reply:
Nevermind, I'm wrong and you also have the lemma already!